lib/Models/loadGeoJson.js

```diff
@@ -0,0 +1,470 @@
+var loadJson = require('../Core/loadJson');
+var PolylineGraphics = require('terriajs-cesium/Source/DataSources/PolylineGraphics');
+var when = require('terriajs-cesium/Source/ThirdParty/when');
+var defaultValue = require('terriajs-cesium/Source/Core/defaultValue');
+var PointGraphics = require('terriajs-cesium/Source/DataSources/PointGraphics');
+var zip = require("terriajs-cesium/Source/ThirdParty/zip");
+
+var standardCssColors = require('../Core/standardCssColors');
+var formatPropertyValue = require('../Core/formatPropertyValue');
+var hashFromString = require('../Core/hashFromString');
+var Reproject = require('../Map/Reproject');
+
+var simpleStyleIdentifiers = ['title', 'description', //
+'marker-size', 'marker-symbol', 'marker-color', 'stroke', //
+'stroke-opacity', 'stroke-width', 'fill', 'fill-opacity'];
+
+// This next function modelled on Cesium.geoJsonDataSource's defaultDescribe.
+function describeWithoutUnderscores(properties, nameProperty) {
+    var html = '';
+    for (var key in properties) {
+        if (properties.hasOwnProperty(key)) {
+            if (key === nameProperty || simpleStyleIdentifiers.indexOf(key) !== -1) {
+                continue;
+            }
+            var value = properties[key];
+            if (typeof value === 'object') {
+                value = describeWithoutUnderscores(value);
+            } else {
+                value = formatPropertyValue(value);
+            }
+            key = key.replace(/_/g, ' ');
+            if (defined(value)) {
+                html += '<tr><th>' + key + '</th><td>' + value + '</td></tr>';
+            }
+        }
+    }
+    if (html.length > 0) {
+        html = '<table class="cesium-infoBox-defaultTable"><tbody>' + html + '</tbody></table>';
+    }
+    return html;
+}
+
+function getJson(entry, deferred) {
+    entry.getData(new zip.Data64URIWriter(), function(uri) {
+        deferred.resolve(loadJson(uri));
+    });
+}
+
+function updateModelFromData(geoJsonItem, geoJson) {
+    // If this GeoJSON data is an object literal with a single property, treat that
+    // property as the name of the data source, and the property's value as the
+    // actual GeoJSON.
+    var numProperties = 0;
+    var propertyName;
+    for (propertyName in geoJson) {
+        if (geoJson.hasOwnProperty(propertyName)) {
+            ++numProperties;
+            if (numProperties > 1) {
+                break; // no need to count past 2 properties.
+            }
+        }
+    }
+
+    var name;
+    if (numProperties === 1) {
+        name = propertyName;
+        geoJson = geoJson[propertyName];
+
+        // If we don't already have a name, or our name is just derived from our URL, update the name.
+        if (
+            !defined(geoJsonItem.name) ||
+            geoJsonItem.name.length === 0 ||
+            nameIsDerivedFromUrl(geoJsonItem.name, geoJsonItem.url)
+        ) {
+            geoJsonItem.name = name;
```

Comment:
You can't directly set traits on a model in the new architecture. The preferred way is to create a loadable stratum for any traits that can be derived from the data or metadata.